Source/EngineClassesTests/GameEngineTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using EngineClasses;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace EngineClasses.Tests
+{
+    [TestClass()]
+    public class GameEngineTests
+    {
+
+        [TestMethod()]
+        public void CreatePlayerTest_CreateNewPlayer_OneNewPlayerWithFourGamePieces()
+        {
+            GameEngine g1 = new GameEngine(new Session(), new GameBoard(), new GameLog());
+            string userName = "Testman";
+            string color = "Red";
+
+            g1.CreatePlayer(userName, color);
+
+            Assert.AreEqual(4, g1.Session.Player.ToList().First().GamePiece.Count());
```

Comment:
I'm a bit unsure what you are testing but this query should be more like:
`g1.Session.Player.First().GamePiece.Count()` the difference is that this will be executed at the SQL server the other in your code